R/util-checkSpec.R

```diff
@@ -54,6 +54,30 @@ CheckSpec <- function(lData, lSpec) {
   allCols <- c()
   missingCols <- c()
   for (strDataFrame in lSpecDataFrames) {
+    chrDataFrameColnames <- colnames(lData[[strDataFrame]])
+    #check modes in data
+    wrongType <- purrr::reduce2(
+      lSpec[[strDataFrame]],
+      names(lSpec[[strDataFrame]]),
+      function(so_far, x, idx) {
+        if (!is.null(x$type) && idx %in% chrDataFrameColnames) {
+          #check if data is the expected mode
+          res <- all(x$type %in% mode(lData[[strDataFrame]][[idx]]))
```

Comment:
[Hadley says not to use mode in Advanced R](https://adv-r.hadley.nz/base-types.html#numeric-type:~:text=Do%20not%20use%20these%20functions), but I think it's the right thing here, since we're using "numeric" for type (`typeof()` is more specific about type). I'm not sure this code is exactly right yet. Will x$type always be length 1? Because I'm almost certain `mode()` is always exactly one value. In that case, this should either be either `x$type == mode(...)`, or `any(x$type == mode(...))` if we want to allow x$type to have multiple alternatives.

Reply:
oh yep- should be `==` not `%in%`. this was leftover from when i was trying this out with `class`, which can have multiple values. will change it to `==`